stdlib/LibCURL_jll/src/LibCURL_jll.jl

```diff
@@ -3,44 +3,21 @@
 ## dummy stub for https://github.com/JuliaBinaryWrappers/LibCURL_jll.jl
 
 baremodule LibCURL_jll
-using Base, Libdl, nghttp2_jll
-Base.Experimental.@compiler_options compile=min optimize=0 infer=false
-
-const PATH_list = String[]
-const LIBPATH_list = String[]
-
+using Base, Libdl, nghttp2_jll, LibSSH2_jll
 export libcurl
 
-# These get calculated in __init__()
-const PATH = Ref("")
-const LIBPATH = Ref("")
-artifact_dir::String = ""
-libcurl_handle::Ptr{Cvoid} = C_NULL
-libcurl_path::String = ""
-
 if Sys.iswindows()
-    const libcurl = "libcurl-4.dll"
+    const libcurl_name = "bin/libcurl-4.dll"
 elseif Sys.isapple()
-    const libcurl = "@rpath/libcurl.4.dylib"
+    const libcurl_name = "lib/libcurl.4.dylib"
 else
-    const libcurl = "libcurl.so.4"
-end
-
-function __init__()
-    global libcurl_handle = dlopen(libcurl)
-    global libcurl_path = dlpath(libcurl_handle)
-    global artifact_dir = dirname(Sys.BINDIR)
-    LIBPATH[] = dirname(libcurl_path)
-    push!(LIBPATH_list, LIBPATH[])
+    const libcurl_name = "lib/libcurl.so.4"
 end
 
-# JLLWrappers API compatibility shims.  Note that not all of these will really make sense.
-# For instance, `find_artifact_dir()` won't actually be the artifact directory, because
-# there isn't one.  It instead returns the overall Julia prefix.
-is_available() = true
-find_artifact_dir() = artifact_dir
-dev_jll() = error("stdlib JLLs cannot be dev'ed")
-best_wrapper = nothing
-get_libcurl_path() = libcurl_path
+const libcurl_path = BundledLazyLibraryPath(libcurl_name)
+const libcurl = LazyLibrary(libcurl_path; dependencies=[libssh2, libnghttp2],
+                            on_load_callback = () -> begin
+                                @assert ccall(dlsym(libcurl, :curl_global_init), UInt32, (Clong,), 0x03) == 0
```

Comment:
Should not be an assert.

Reply:
Thanks for the feedback; this PR is incredibly old, but I did refer to it when I made my recent LinearAlgebra PR, so it's still useful to comment on here.

This particular assert came from [this `@check` macro in `Downloads`](https://github.com/JuliaLang/Downloads.jl/blob/1199e9039dcce6072601c3f29cea8da4a0acbff6/src/Curl/Curl.jl#L91C6-L91C11), but I see now it simply called `@error`, so I should probably do the same here.